src/langchain_google_cloud_sql_pg/postgresql_engine.py

```diff
@@ -0,0 +1,189 @@
+        thread = Thread(target=loop.run_forever, daemon=True)
+        thread.start()
+        coro = cls.afrom_instance(project_id, region, instance, database)
+        return asyncio.run_coroutine_threadsafe(coro, loop).result()
+
+    @classmethod
+    async def _create(
+        cls,
+        project_id: str,
+        region: str,
+        instance: str,
+        database: str,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        thread: Optional[Thread] = None,
+    ) -> PostgreSQLEngine:
+        credentials, _ = google.auth.default(
+            scopes=["https://www.googleapis.com/auth/userinfo.email"]
+        )
+        iam_database_user = await _get_iam_principal_email(credentials)
+        if cls._connector is None:
+            cls._connector = await create_async_connector()
+
+        # anonymous function to be used for SQLAlchemy 'creator' argument
+        def getconn() -> asyncpg.Connection:
+            conn = cls._connector.connect_async(  # type: ignore
+                f"{project_id}:{region}:{instance}",
+                "asyncpg",
+                user=iam_database_user,
+                db=database,
+                enable_iam_auth=True,
+            )
+            return conn
+
+        engine = create_async_engine(
+            "postgresql+asyncpg://",
+            async_creator=getconn,
+        )
+        return cls(engine, loop, thread)
+
+    @classmethod
+    async def afrom_instance(
+        cls,
+        project_id: str,
+        region: str,
+        instance: str,
+        database: str,
+    ) -> PostgreSQLEngine:
+        return await cls._create(project_id, region, instance, database)
+
+    async def aexecute(self, query: str):
+        async with self._engine.connect() as conn:
+            await conn.execute(text(query))
+            await conn.commit()
+
+    async def afetch(self, query: str):
+        async with self._engine.connect() as conn:
+            result = await conn.execute(text(query))
+            result_map = result.mappings()
+            result_fetch = result_map.fetchall()
+
+        return result_fetch
+
+    def run_as_sync(self, coro: Awaitable[T]):  # TODO: add return type
+        if not self._loop:
+            raise Exception("Engine was initialized async.")
+        return asyncio.run_coroutine_threadsafe(coro, self._loop).result()
+
+    async def ainit_vectorstore_table(
+        self,
+        table_name: str,
+        vector_size: int,
+        content_column: str = "content",
+        embedding_column: str = "embedding",
+        metadata_columns: List[Column] = [],
+        id_column: str = "langchain_id",
+        overwrite_existing: bool = False,
+        store_metadata: bool = True,
+    ) -> None:
+        await self.aexecute("CREATE EXTENSION IF NOT EXISTS vector")
+
+        if overwrite_existing:
+            await self.aexecute(f"DROP TABLE {table_name}")
+
+        query = f"""CREATE TABLE IF NOT EXISTS {table_name}(
```

Comment:
nit: potential footgun
```suggestion
        query = f"""CREATE TABLE {table_name}(
```